src/cryptonote_core/service_node_list.h

```diff
@@ -521,17 +517,23 @@ namespace service_nodes
       quorum_manager quorums;
     };
 
-    std::deque<quorums_by_height>             m_old_quorum_states; // Store all old quorum history only if run with --store-full-quorum-history
-    state_set                                 m_state_history; // Store state_t's from MIN(2nd oldest checkpoint | height - DEFAULT_SHORT_TERM_STATE_HISTORY) up to the block height
-    state_set                                 m_state_archive; // Store state_t's where ((height < m_state_history.first()) && (height % STORE_LONG_TERM_STATE_INTERVAL))
-    state_t                                   m_state;
-    std::unordered_map<crypto::hash, state_t> m_alt_state;
-    std::unordered_map<crypto::public_key, proof_info> m_proofs;
-
-    bool                                   m_state_added_to_archive;
-    data_for_serialization                 m_cache_long_term_data;
-    data_for_serialization                 m_cache_short_term_data;
-    std::string                            m_cache_data_blob;
+    struct
+    {
+      /// Maps x25519 pubkeys to registration pubkeys + last block seen value (used for expiry)
+      std::unordered_map<crypto::x25519_public_key, std::pair<crypto::public_key, time_t>> x25519_to_pub;
+      time_t                                                                               x25519_map_last_pruned = 0;
+      std::deque<quorums_by_height>                                                        old_quorum_states; // Store all old quorum history only if run with --store-full-quorum-history
+      state_set                                                                            state_history; // Store state_t's from MIN(2nd oldest checkpoint | height - DEFAULT_SHORT_TERM_STATE_HISTORY) up to the block height
+      state_set                                                                            state_archive; // Store state_t's where ((height < m_state_history.first()) && (height % STORE_LONG_TERM_STATE_INTERVAL))
+      std::unordered_map<crypto::hash, state_t>                                            alt_state;
+      std::unordered_map<crypto::public_key, proof_info>                                   proofs;
```

Comment:
I think proofs is similar in that we don't want to clear it.

Reply:
Added